lib/co2_offset/converters/beef.ex

```diff
@@ -4,14 +4,16 @@ defmodule Co2Offset.Converters.Beef do
   """
   @co2_per_kg 34.6
 
-  @spec convert_and_structure({float(), list()}) :: {float(), nonempty_maybe_improper_list()}
-  def convert_and_structure({co2_amount, acc}) when is_float(co2_amount) and is_list(acc) do
-    kg = co2_amount |> convert(:kg_from_co2)
-
-    {
-      co2_amount,
-      [structured_info(kg, co2_amount) | acc]
-    }
+  @spec convert_and_structure(%{optional(any) => any, co2: co2}) :: %{
+          optional(any) => any,
+          co2: co2,
+          beef: %{kg: float()}
+        }
+        when co2: float
+  def convert_and_structure(acc) when is_map(acc) do
```

Comment:
is it possible to move this code from the `/converters/*` to some kind of module ? coz this looks the same for all converters

Reply:
there are not the same :( . Every kind of this `converter` is using different formatters and converters itself. The shared module can have only one two common methods: `convert_and_structure` and `structure` which should somehow infer the type of conversion unit(`kg`) and converter(`kg_from_co2`).

It will have so bad readability because of metaprogramming that i will prefer to duplicate the code.